tests/integration/models/multifidelity/test_multifidelity_models.py

```diff
@@ -0,0 +1,102 @@
+import numpy as np
+import numpy.testing as npt
+import tensorflow as tf
+
+import trieste
+from trieste.data import (
+    add_fidelity_column,
+    check_and_extract_fidelity_query_points,
+    split_dataset_by_fidelity,
+)
+from trieste.models.gpflow import GaussianProcessRegression
+from trieste.models.gpflow.builders import build_gpr, build_multifidelity_autoregressive_models
+from trieste.models.gpflow.models import MultifidelityAutoregressive
+from trieste.objectives.utils import mk_observer
+from trieste.types import TensorType
+
+
+def noisy_linear_multifidelity(x: TensorType) -> TensorType:
+
+    x_input, x_fidelity = check_and_extract_fidelity_query_points(x)
+
+    f = 0.5 * ((6.0 * x_input - 2.0) ** 2) * tf.math.sin(12.0 * x_input - 4.0) + 10.0 * (
+        x_input - 1.0
+    )
+    f = f + x_fidelity * (f - 20.0 * (x_input - 1.0))
+    noise = tf.random.normal(f.shape, stddev=1e-1, dtype=f.dtype)
+    f = tf.where(x_fidelity > 0, f + noise, f)
+    return f
+
+
+def test_multifidelity_autoregressive_results_close() -> None:
+
+    input_dim = 1
+    lb = np.zeros(input_dim)
+    ub = np.ones(input_dim)
+    n_fidelities = 4
+
+    input_search_space = trieste.space.Box(lb, ub)
+    n_samples_per_fidelity = [
+        2 ** ((n_fidelities - fidelity) + 1) + 3 for fidelity in range(n_fidelities)
+    ]
+
+    xs = [tf.linspace(0, 1, samples)[:, None] for samples in n_samples_per_fidelity]
+    initial_samples_list = [tf.concat([x, tf.ones_like(x) * i], 1) for i, x in enumerate(xs)]
+    initial_sample = tf.concat(initial_samples_list, 0)
+    observer = mk_observer(noisy_linear_multifidelity)
+    initial_data = observer(initial_sample)
+
+    data = split_dataset_by_fidelity(initial_data, n_fidelities)
+
+    gprs = [
+        GaussianProcessRegression(
+            build_gpr(
+                data[fidelity], input_search_space, likelihood_variance=1e-6, kernel_priors=False
+            )
+        )
+        for fidelity in range(n_fidelities)
+    ]
+
+    model = MultifidelityAutoregressive(gprs)
+
+    model.update(initial_data)
+    model.optimize(initial_data)
+
+    test_xs = tf.linspace(0, 1, 11)[:, None]
+    test_xs_w_fid = add_fidelity_column(test_xs, fidelity=3)
+
+    predictions = model.predict(test_xs_w_fid)[0]
+    gt_obs = observer(test_xs_w_fid).observations
+
+    npt.assert_allclose(predictions, gt_obs, rtol=0.20)
+
+
+def test_multifidelity_autoregressive_gets_expected_rhos() -> None:
```

Comment:
We might need a few more tests for the AR model predict: 1) we should test that predicting at LF amounts to using LF predict_f. Then we could create dummy models and check that predict does what is expected, e.g. the HF predict reverts to the HF predict_f predict if rho is zero or if the LF model is flat, or it reverts to the LF prediction if the HF one is flat and zero, etc.